docs/docs/09-rfcs/2023-03-08-wing-preview-environment.md

```diff
@@ -0,0 +1,282 @@
+For each test run, a new simulator will be created and will be destroyed upon test completion.
+The test results will be available in the PR comment for each entry point.
+
+#### Running Tests Manually
+
+Running tests on your environment is easy using the Wing Console tests UI.
+Each test will reload the simulator to create a clean environment.
+
+Once the test is over it will **not** clean up the simulator data, so you can debug the environment and share outputs with your teammates.
+
+#### User-Controlled Environments termination
+
+Closing a PR will terminate the corresponding preview environments.
+The PR comment will indicate the termination of each preview environment and the urls won't be valid anymore.
+
+#### Self-Cleaning Environments Mechanism
+
+To ensure efficient resource utilization:
+
+1. Preview environments associated with stale PRs (without access or code changes for over 30 days) will automatically deactivate.
+2. The PR's preview environment comment will indicate its inactive status.
+3. Changes in code will trigger redeployment for all related preview environments in this PR.
+
+### Analytics and Logs
+
+To improve the product we are collecting data and monitoring the system.
+
+1. We collect preview environments usage analytics only. We won't collect anything about your code and project.
+2. OpenTelemetry is used for collecting metrics, logs, traces.
+3. Datadog is used for monitoring the system.
+
+### Documentation
+
+Developers reading our docs can easily sign up to Wing Cloud and start using its Preview Environments feature in their repository.
+In our docs we have the following information:
+
+1. What is Wing Cloud Preview Environments
+2. Getting started instructions
+3. How can I configure multiple entry points
+4. Populate environment with initial data
+5. Running tests
+
+### Security
+
+1. The preview environments shouldn’t be able to access files from other preview environments
+2. Only Wing Cloud signed up developers with write access to the repository can create preview environments
+
+## NON Requirements
+
+1. Preview environments will require authentication to view.
+2. Assign users to Teams.
+3. enable preview environments viewers to add comments in the environment itself
+
+## Technical Design
+
+### Technology
+
+- https://wing.cloud for the infrastructure (although it may prove a challenge due to time constrains)
+- AWS for the infrastructure
+- Node.js 18 for the backend
+- https://github.com/probot/probot for the GitHub integration
+- https://fly.io for the preview environments
+- https://react.dev/ for the frontend
+
+### Architecture
+
+- A GitHub app
+- A Cloud.Function that will be triggered by GitHub events, such as `pull_request.synchronize` or `pull_request.closed`
+- A Flyio account to deploy the preview environments
+- A Cloud.Table to store different data:
+  - The registered users
+  - The Wing Cloud projects and their config
```

Comment:
is it relevant for the mvp? (next line as well)

Reply:
Not anymore since we don't want configuration yet. I'll rephrase it.